MALA method (#29)
TuringLang/AdvancedMH.jl: README.md

# Sample 4 chains from the posterior.
chain = psample(model, RWMH(init_params), 100000, 4; param_names=["μ","σ"], chain_type=Chains)
```

## Metropolis-adjusted Langevin algorithm (MALA)

AdvancedMH.jl also offers an implementation of [MALA](https://en.wikipedia.org/wiki/Metropolis-adjusted_Langevin_algorithm) if the `ForwardDiff` and `DiffResults` packages are available. 

> A set of `init_params` are required. `x->` within the `MALA` struct takes the gradient computed at the current sample.

Reviewer: Maybe it would make sense to implement `propose(rng::Random.AbstractRNG, ::MALA, model) = error("please specify initial parameters")` since otherwise users will only get a dispatch error.

I think it might also be better to explain the constructor instead of the struct (which should not matter for users):
```suggestion
A `MALA` sampler can be constructed by `MALA(proposal)` where `proposal` is a function that
takes the gradient computed at the current sample. It is required to specify an initial sample `init_params` when calling `sample`.
```

(BTW: It's good that it is the first argument since this allows one to write
```julia
sampler = MALA() do x
  # some complicated calculations
  ....
  return proposal
end
```)

```julia
# Import the package.
using AdvancedMH
using Distributions
using MCMCChains

> using LinearAlgebra

Reviewer: ```suggestion
```

using DiffResults
using ForwardDiff

# Generate a set of data from the posterior we want to estimate.
data = rand(Normal(0, 1), 30)

# Define the components of a basic model.
insupport(θ) = θ[2] >= 0
dist(θ) = Normal(θ[1], θ[2])
density(θ) = insupport(θ) ? sum(logpdf.(dist(θ), data)) : -Inf

# Construct a DensityModel.
model = DensityModel(density)

> #MvNormal covariance
> Sigma = 1e-2 * I(2)
> 
> # Set up our sampler with a joint multivariate Normal proposal.
> spl = MALA(x-> MvNormal((1/2) * Sigma * x, Sigma))

Reviewer: This does not work on Julia 1.0 (see my comments for the tests as well). Instead you can write:
```suggestion
# Set up the sampler with a multivariate Gaussian proposal.
sigma = 1e-1
spl = MALA(x -> MvNormal((sigma^2 / 2) .* x, sigma))
```
